crates/env/src/engine/on_chain/pallet_revive.rs

```diff
@@ -553,6 +553,25 @@ fn call_storage_precompile(
     )
 }
 
+/// Simple decoder for a Solidity `bytes` type.
+///
+/// Returns the number of bytes written to `out`.
+fn decode_bytes(input: &[u8], out: &mut [u8]) -> usize {
+    let mut buf = [0u8; 4];
+    buf[..].copy_from_slice(&input[28..32]);
+    let offset = u32::from_be_bytes(buf) as usize;
+
+    let mut buf = [0u8; 4];
+    buf[..].copy_from_slice(&input[60..64]);
+    let bytes_len = u32::from_be_bytes(buf) as usize;
+
+    // we start decoding at the start of the payload.
+    // the payload starts at the `len` word here:
+    // `bytes = offset (32 bytes) | len (32 bytes) | data`
+    out[..bytes_len].copy_from_slice(&input[32 + offset..32 + offset + bytes_len]);
```

Comment:
```suggestion
    out[..bytes_len].copy_from_slice(&input[64..64 + bytes_len]);
```
starting index for data/payload is always 64 for the encoding of a single bytes param or return type